src/main/java/org/mockbukkit/mockbukkit/block/data/BlockDataMock.java

```diff
@@ -45,6 +55,27 @@
 public class BlockDataMock implements BlockData
 {
 
+	/**
+	 * This factory tries to create the block data from a material {@link Tag}.
+	 */
+	private static final Map<Tag<Material>, Function<Material, BlockDataMock>> FACTORIES_BY_TAGS = Map.of(
+			Tag.CANDLES, CandleDataMock::new,
+			Tag.RAILS, RailDataMock::new
+	);
```

Comment:
I was a bit confused here initially; I thought you would replace all the functionallity of attemptMockByTag, but that's not the case. I'm fine with both implementations here, really. But I think we have to choose which one to have, to reduce maintanence and confusion

Reply:
Nice catch, thanks. I'll update it